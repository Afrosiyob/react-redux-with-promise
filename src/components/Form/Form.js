import React, { useState } from "react";

const Form = (props) => {
  const { username, password } = props;

  const [values, setValues] = useState({
    username: username || "",
    password: password || "",
  });

  const inputChange = (e) => {
    const { name, value } = e.target;
    setValues((prevValues) => ({ ...prevValues, [name]: value }));
  };

  const handleSubmit = () => {
    console.log(values);
  };

  return (
    <>
      <form>
        <label htmlFor="username"> Username: </label>
        <input
          type="text"
          name="username"
          id="username"
          onChange={inputChange}
        />
        <br />
        <label htmlFor="password"> Password: </label>
        <input
          type="password"
          name="password"
          id="password"
          onChange={inputChange}
        />
        <br />

        <button onClick={handleSubmit}> Submit </button>
      </form>
    </>
  );
};

export default Form;
