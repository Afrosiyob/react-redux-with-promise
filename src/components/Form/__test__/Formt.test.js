import { screen, render } from "@testing-library/react";
import Form from "../Form";

import renderer from "react-test-renderer";

test("Form", () => {
  render(<Form />);
  screen.debug();

  const labelUsername = screen.getByText(/username:/i);
  screen.getByRole("textbox", {
    name: /username:/i,
  });
  expect(labelUsername).toHaveTextContent(/username/i);

  const labelPassword = screen.getByText(/password:/i);

  expect(labelPassword).toHaveTextContent(/password/i);

  const tree = renderer.create(<Form />).toJSON();
  expect(tree).toMatchSnapshot();
});
