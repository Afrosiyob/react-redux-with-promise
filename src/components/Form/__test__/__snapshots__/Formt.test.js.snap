// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form 1`] = `
<form>
  <label
    htmlFor="username"
  >
     Username: 
  </label>
  <input
    id="username"
    name="username"
    onChange={[Function]}
    type="text"
  />
  <br />
  <label
    htmlFor="password"
  >
     Password: 
  </label>
  <input
    id="password"
    name="password"
    onChange={[Function]}
    type="password"
  />
  <br />
  <button
    onClick={[Function]}
  >
     Submit 
  </button>
</form>
`;
